### PR TITLE
Add battlenet

### PR DESCRIPTION
**Pull request by kharf**
Basically copied the star citizen config. Not sure about the GL settings, I think its for nvidia. I cant really test it as I am using AMD. I have been using this setup for a couple of days now and it works flawlessly.

### CODE REVIEW

**Comment by kharf** [comment]
just saw that my commit doesn`t adhere to the common style. Should I amend it?

**Comment by NotAShelf** [comment]
you can `reword` in git rebase interactive

**Comment by kharf** [comment]
> you can `reword` in git rebase interactive

yea thats basically the same. Will do so

File: pkgs/battlenet/default.nix [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  wineFlags ? "",
+  pname ? "battlenet",
+  location ? "$HOME/Games/battlenet",
+  tricks ? ["arial" "tahoma" "win10"],
+  preCommands ? "",
+  postCommands ? "",
+  pkgs,
+}: let
+  src = pkgs.fetchurl rec {
+    url = "https://www.battle.net/download/getInstallerForGame?os=win&version=LIVE&gameProgram=BATTLENET_APP";
```
Comment (kharf) [inline pkgs/battlenet/default.nix:19]: still figuring out how to get this reproducible ... havent found a versioned link

Comment (kharf) [reply pkgs/battlenet/default.nix:19]: @fufexan do you know if there is a recommended or common way to approach this?

Comment (fufexan) [reply pkgs/battlenet/default.nix:19]: I don't. I've looked in the lutris Battle.net script and it's the same URL. If it's not deterministic on their end, there's not much we can do. Maybe there are some other parameters that we don't know about.

Comment (fufexan) [reply pkgs/battlenet/default.nix:19]: I just searched for the versions and this URL seems to yield deterministic results, but we'll have to update it periodically in some way.
https://downloader.battle.net/download/getInstallerForGame?os=win&gameProgram=BATTLENET_APP&version=2.26.0.14494

Comment (fufexan) [reply pkgs/battlenet/default.nix:19]: From what I can see, we can use https://blizztrack.com/swagger/index.html in an update script, but I don't have time to fiddle with it right now.

Comment (kharf) [reply pkgs/battlenet/default.nix:19]: Ill see what I can do, thank you. Probably will take some time, because of minimal free time

**Pull request closed without merging**